### PR TITLE
[sdk/go]: Add support for getResource to Go SDK

### LINKED ISSUE #5797: [sdk/go] Add support for getResource
Go implementation of https://github.com/pulumi/pulumi/pull/5694

### PR DESCRIPTION
**Pull request by lblackstone**
Fix #5797

### CODE REVIEW

**Review by pgavlin — APPROVED** [review]
Couple of suggestions, but LGTM overall.

File: sdk/go/pulumi/context.go [hunk 1]
```diff
@@ -414,6 +413,52 @@ func (ctx *Context) RegisterResource(
...
+	// If there were any failures from the provider, return them.
+	if len(resp.Failures) > 0 {
+		logging.V(9).Infof("Invoke(%s, ...): success: w/ %d failures", tok, len(resp.Failures))
+		var ferr error
+		for _, failure := range resp.Failures {
+			ferr = multierror.Append(ferr,
+				fmt.Errorf("%s invoke failed: %s (%s)", tok, failure.Reason, failure.Property))
+		}
+		return nil, ferr
+	}
```
[the remark this answers is not in the document]
Comment (lblackstone) [reply sdk/go/pulumi/context.go:450]: These failure logs are related to the `getResource` invoke, so I left them in.

File: sdk/go/pulumi/resource.go [hunk 2]
```diff
@@ -192,6 +194,8 @@ type invokeOptions struct {
 	Parent Resource
 	// Provider is an optional provider resource to use for this invoke.
 	Provider ProviderResource
+	// URN is an optional URN of a previously-registered resource of this type to read from the engine for this invoke.
+	URN string
```
Comment (justinvp) [inline sdk/go/pulumi/resource.go:198]: Why is this needed on `invokeOptions`?

Comment (lblackstone) [reply sdk/go/pulumi/resource.go:198]: Hmm, I guess this may be a copy-paste error. It doesn't seem like it's needed, so I'll rework this to remove it.
